### PR TITLE
Upgrade go-github from v29 to v31

### PR DESCRIPTION
**Pull request by LukeZielinski**
The intent is to pull in the new oauth implementation.

Ran the following commands:
$ find . -name *.go | xargs sed -i 's|github.com/google/go-github/v29/github|github.com/google/go-github/v31/github|'
$ go get -t -u -v -d ./...
$ go mod tidy

### CODE REVIEW

File: shared/triage_metadata.go [hunk 1]
```diff
@@ -108,10 +108,11 @@ func (tm triageMetadata) getCommitBranchRef() (ref *github.Reference, err error)
 func (tm triageMetadata) getTree(ref *github.Reference, triagedMetadataMap map[string][]byte) (tree *github.Tree, err error) {
 	client := tm.githubClient
 
-	entries := []github.TreeEntry{}
+	entries := []*github.TreeEntry{}
 	for folderPath, content := range triagedMetadataMap {
 		dest := GetMetadataFilePath(folderPath)
-		entries = append(entries, github.TreeEntry{Path: github.String(dest), Type: github.String("blob"), Content: github.String(string(content)), Mode: github.String("100644")})
+		newEntry := github.TreeEntry{Path: github.String(dest), Type: github.String("blob"), Content: github.String(string(content)), Mode: github.String("100644")}
```
Comment (LukeZielinski) [inline shared/triage_metadata.go:114]: Not too sure whether newEntry is needed here - I was getting "literal is not a type" errors when trying to inline creation of this TreeEntry point inside the call to append, and I couldn't really make sense of it. Open to suggestions.

Comment (Hexcles) [reply shared/triage_metadata.go:114]: That's interesting. Is that the full error message?

For example, this works: https://play.golang.org/p/aNPerJmBrRS

Comment (LukeZielinski) [reply shared/triage_metadata.go]: Ahh, ok, I was just thinking of this in C++ style (where taking the address of a stack-allocated variable is bad news). I was calling `append(entries, new(github.TreeEntry{...` which was throwing that error. Updated to your suggestion, which seems to work.

Comment (Hexcles) [reply shared/triage_metadata.go]: Right, `new` only takes a type. Stack and heap are basically hidden away in Golang by GC and the memory model is closer to Java. Even if the object was indeed created on the stack, GC would be responsible for moving it out and updating all referencing pointers.

Just keep in mind don't cast the pointer to untyped `uintptr`, at which point there'll be no guarantees.

**Comment by KyleJu** [comment]
This should fix https://github.com/web-platform-tests/wpt.fyi/issues/1834

**Pull request merged**